subscription-cache/src/lib.rs

```diff
@@ -0,0 +1,101 @@
+/********************************************************************************
+ * Copyright (c) 2024 Contributors to the Eclipse Foundation
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information regarding copyright ownership.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Apache License Version 2.0 which is available at
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ ********************************************************************************/
+
+use async_std::sync::Mutex;
+use protobuf::MessageField;
+use std::collections::{HashMap, HashSet};
+use std::hash::{Hash, Hasher};
+use up_rust::core::usubscription::{
+    EventDeliveryConfig, FetchSubscriptionsResponse, SubscribeAttributes, SubscriberInfo,
+    SubscriptionStatus,
+};
+use up_rust::UUri;
+
+pub type SubscribersMap = Mutex<HashMap<MessageField<UUri>, HashSet<SubscriptionInformation>>>;
```

Comment:
Is there a reason to include the MessageField types, instead of directly going for the contained types?